### PR TITLE
RI-6067 implemented reset/start/stop rdi functionality

### PR DESCRIPTION
**Pull request by kchepikava**

### CODE REVIEW

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/reset-pipeline-button/ResetPipelineButton.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import { EuiButton, EuiSpacer, EuiToolTip } from '@elastic/eui'
+import cx from 'classnames'
+import ResetIcon from 'uiSrc/assets/img/rdi/reset.svg?react'
+import styles from './styles.module.scss'
+
+export interface PipelineButtonProps {
+  onClick: () => void;
+  disabled: boolean;
+  loading: boolean;
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/reset-pipeline-button/ResetPipelineButton.tsx:10]: please, remove semicolons

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/start-pipeline-button/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,30 @@
+.pipelineBtn {
+  border: 1px solid var(--euiColorSecondary) !important;
+
+  // :global(.euiButton__text) {
+  //   font-weight: 400 !important;
+  //   font-size: 14px !important;
+  //   line-height: 16.8px !important;
+  // }
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/start-pipeline-button/styles.module.scss:8]: please, remove commented code

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/reset-pipeline-button/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,30 @@
+.pipelineBtn {
+  border: 1px solid var(--euiColorSecondary) !important;
+
+  // :global(.euiButton__text) {
+  //   font-weight: 400 !important;
+  //   font-size: 14px !important;
+  //   line-height: 16.8px !important;
+  // }
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/reset-pipeline-button/styles.module.scss:8]: please, remove commented code

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/stop-pipeline-button/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
+.stopPipelineBtn {
+  border: 1px solid var(--euiColorSecondary) !important;
+
+  // :global(.euiButton__text) {
+  //   font-weight: 400 !important;
+  //   font-size: 14px !important;
+  //   line-height: 16.8px !important;
+  // }
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/stop-pipeline-button/styles.module.scss:8]: the same

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/stop-pipeline-button/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,26 @@
+.stopPipelineBtn {
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/buttons/stop-pipeline-button/styles.module.scss:1]: if we use the same styles for several buttons, we can move it to one file

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/pipeline-actions/PipelineActions.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+  // })
+
+  // it('should call onSubmit and close popover', () => {
+  //   render(<PipelineActions {...mockedProps} />)
+
+  //   fireEvent.click(screen.getByTestId('deploy-rdi-pipeline'))
+  //   fireEvent.click(screen.getByTestId('deploy-confirm-btn'))
+
+  //   expect(mockHandleSubmit).toBeCalled()
+  // })
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/pipeline-actions/PipelineActions.spec.tsx:162]: please, remove commented code

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/pipeline-actions/PipelineActions.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
+import React from 'react'
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+
+import { getPipelineStatusAction, rdiPipelineResetSelector, rdiPipelineSelector, rdiPipelineStartSelector, rdiPipelineStopSelector, resetPipelineAction, startPipelineAction, stopPipelineAction } from 'uiSrc/slices/rdi/pipeline'
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/pipeline-actions/PipelineActions.tsx:9]: please, refactor to have correct max line length

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/rdi-config-file-action-menu/RdiConfigFileActionMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  EuiButtonIcon,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui'
+import cx from 'classnames'
+import threeDots from 'uiSrc/assets/img/rdi/three_dots.svg?react'
+import uploadIcon from 'uiSrc/assets/img/rdi/upload.svg?react'
+import FetchPipelinePopover from '../fetch-pipeline-popover'
+import UploadModal from '../../../../../pipeline-management/components/upload-modal/UploadModal'
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/rdi-config-file-action-menu/RdiConfigFileActionMenu.tsx:13]: to many `../` its better to use uiSrc,
also you could add index.ts to have less path

File: redisinsight/ui/src/pages/rdi/instance/components/header/components/rdi-config-file-action-menu/RdiConfigFileActionMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+import Download from '../../../download/Download'
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/instance/components/header/components/rdi-config-file-action-menu/RdiConfigFileActionMenu.tsx:14]: also, here could you add index.ts

File: redisinsight/ui/src/pages/rdi/statistics/StatisticsPage.tsx [hunk 10]
```diff
@@ -99,54 +98,52 @@ const StatisticsPage = () => {
...
-                  onChangeAutoRefresh('clients', enableAutoRefresh, refreshRate)}
-              />
-            </>
-          )}
+    <div className={styles.pageContainer}>
+      <div className={styles.bodyContainer}>
+        {pageLoading && (
+        <div className={styles.cover}>
+          <EuiLoadingSpinner size="xl" />
         </div>
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/pages/rdi/statistics/StatisticsPage.tsx:106]: please, add tabulation

File: redisinsight/ui/src/slices/rdi/pipeline.ts [hunk 11]
```diff
@@ -366,3 +433,86 @@ export function getPipelineStatusAction(
     }
   }
 }
+
+export function stopPipelineAction(
+  rdiInstanceId: string,
+  callback: (result: IActionPipelineResultProps) => void,
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/rdi/pipeline.ts:439]: it's better to split to onSuccess and onFinish

File: redisinsight/ui/src/slices/rdi/pipeline.ts [hunk 12]
```diff
@@ -366,3 +433,86 @@ export function getPipelineStatusAction(
...
+      dispatch(addErrorNotification(parsedError))
+      dispatch(stopPipelineFailure(errorMessage))
+      callback?.({ success: false, error: errorMessage })
+    }
+  }
+}
+
+export function startPipelineAction(
+  rdiInstanceId: string,
+  callback: (result: IActionPipelineResultProps) => void,
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/rdi/pipeline.ts:466]: the same

File: redisinsight/ui/src/slices/rdi/pipeline.ts [hunk 13]
```diff
@@ -366,3 +433,86 @@ export function getPipelineStatusAction(
...
+      dispatch(addErrorNotification(parsedError))
+      dispatch(startPipelineFailure(errorMessage))
+      callback?.({ success: false, error: errorMessage })
+    }
+  }
+}
+
+export function resetPipelineAction(
+  rdiInstanceId: string,
+  callback: (result: IActionPipelineResultProps) => void,
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/rdi/pipeline.ts:493]: the same

File: redisinsight/ui/src/slices/rdi/pipeline.ts [hunk 14]
```diff
@@ -72,6 +90,42 @@ const rdiPipelineSlice = createSlice({
...
+      state.reset.error = ''
+    },
+    resetPipelineSuccess: (state) => {
+      state.reset.loading = false
+      state.reset.error = ''
+    },
+    resetPipelineFailure: (state, { payload }: PayloadAction<string>) => {
+      state.reset.loading = false
+      state.reset.error = payload
+    },
```
Comment (romansergeenkosofteq) [inline redisinsight/ui/src/slices/rdi/pipeline.ts:128]: a lot similar actions, could you think how to have one? Maybe add some variables (isStarted)? Do we use all these errors at the same time?

**Review by romansergeenkosofteq — DISMISSED** [review]
approve FE

**Comment by zalenskiSofteq** [comment]
approve BE

**Review by zalenskiSofteq — APPROVED** [review]

**Review by mariasergeenko — APPROVED** [review]

**Pull request merged**